Fixed the regression bug for aggregates with no window.

hjafarpour opened the pull request.

apurvam commented:
What bug did you fix? When was the regression introduced? Also, can we add unit tests? I see only integration tests for the fix.

apurvam commented:
Also worth including in the description: how does the bug manifest? What are the symptoms? Do tests fail? Do queries give wrong results sometimes?

hjafarpour commented:
The but was introduced when we switched to the new Streams API in 1.0. The state store serde was not set correctly. This resulted in query failure because of incorrect serde settings.
The tests did not catch it since we did not have a test with GROUP BY without window. I added an integration test to capture this case too.

dguy commented on ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKGroupedStream.java:
```diff
@@ -61,13 +61,13 @@ public SchemaKTable aggregate(final Initializer initializer,
     if (windowExpression != null) {
       final Materialized<String, GenericRow, ?> materialized
-          = Materialized.<String, GenericRow, WindowStore<Bytes, byte[]>>as(storeName)
+          = Materialized.<String, GenericRow, WindowStore<String, GenericRow>>as(storeName)
```
This is actually incorrect. It should be `WindowStore<Bytes, byte[]>` the underlying store is always of type `<Bytes, byte[]>` now.

It should be:
```
Materialized.<String, GenericRow, WindowStore<Bytes, byte[]>>as(storeName)
   .withKeySerde(Serdes.String())
   .withValueSerde(topicValueSerDe)

hjafarpour replied:
@dguy thanks for the correction. Fixed it.

dguy commented on ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java:
```diff
@@ -69,6 +69,52 @@ public void after() throws Exception {
+
+    Set<String> topicBeforeCleanup = topicClient.listTopicNames();
+
+    assertThat("Expected to have 5 topics instead have : " + topicBeforeCleanup.size(),
```
`assertThat("..", topicBeforeCleanup.size(), equalTo(5))`

hjafarpour replied:
Done.

dguy commented on ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java:
```diff
@@ -69,6 +69,52 @@ public void after() throws Exception {
+    queryMetadata.close();
+    Set<String> topicsAfterCleanUp = topicClient.listTopicNames();
+
+    assertThat("Expected to see 3 topics after clean up but seeing " + topicsAfterCleanUp.size
```
`assertThat("..", topicsAfterCleanup.size(), equalTo(3))`

hjafarpour replied:
Done.
